misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/MockAnnotationProcessor.java

```diff
@@ -179,6 +179,13 @@ public void process(FunctionNode functionNode, List<AnnotationAttachmentNode> an
             String functionName = functionNode.getName().getValue();
 
             if (MOCK_ANNOTATION_NAME.equals(annotationName)) {
+                // Check if the package belongs to a single file project
+                if (SINGLE_FILE_PACKAGE_NAME.equals(packageName)) {
+                    diagnosticLog.logDiagnostic(
+                            DiagnosticSeverity.ERROR, (ModuleDescriptor) null, attachmentNode.getPosition(),
+                            "function mocking is not supported with standalone Ballerina files");
```

Comment:
```suggestion
                            "compile-time function mocking is not supported with standalone Ballerina files");
```

Reply:
Should the error message for both legacy and new function mocking be changed to match this?